server/routes/signup.js

```diff
@@ -12,14 +11,22 @@ router.post('/signup', function (req, res) {
     school: req.body.school
   }, process.env.TOKEN_SECRET)
 
+  if (req.body.school === '' || req.body.name === '') {
```

Comment:
This will not crash the server. The name can be anything up to the student just as long as it's not empty. 
I've taken the req.body.school out because the school name check is implemented elsewhere